ocaml-lsp-server/src/ocamlformat_rpc.ml

```diff
@@ -0,0 +1,172 @@
+open Import
+
+type process =
+  { pid : int
+  ; input : in_channel
+  ; output : out_channel
+  ; io_thread : Scheduler.thread
+  ; client : Ocamlformat_rpc_lib.client
+  }
+
+type state =
+  | Binary_not_found
+  | Stopped
+  | Running of process
+  | Errored
+
+type t = state ref
+
+let supported_versions = [ "v1" ]
+
+let pick_client ~pid input output io_thread =
+  let open Fiber.O in
+  match
+    Scheduler.async io_thread (fun () ->
+        Ocamlformat_rpc_lib.pick_client ~pid input output supported_versions)
+  with
+  | Error `Stopped -> Fiber.return @@ Error (`Msg "Thread stopped")
+  | Ok res -> (
+    let+ res = Scheduler.await_no_cancel res in
+    match res with
+    | Ok s -> s
+    | Error e -> Exn_with_backtrace.reraise e)
+
+let configure process =
+  let open Fiber.O in
+  let { io_thread; client; _ } = process in
+  (* We ask for 64 columns formatting as this appear to be the maximum size of
+     VScode popups. TODO We should probably allow some flexibility for other
+     editors that use the server. *)
+  match
+    Scheduler.async io_thread (fun () ->
+        match
+          Ocamlformat_rpc_lib.config
+            [ ("module-item-spacing", "compact"); ("margin", "64") ]
+            client
+        with
+        | Error (`Msg msg) ->
+          Log.log ~section:"ocamlformat" (fun () ->
```

Comment:
I'd move this back to the main thread. it's better to write log messages from a single thread if possible.